docs/sdks/client-sdks/javascript/assignments.mdx

```diff
@@ -0,0 +1,411 @@
+import * as EppoSdk from "@eppo/js-client-sdk";
+
+const client = EppoSdk.getInstance();
+
+const flagKey = "flag-key-123";
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = false; 
+
+const variant = client.getBooleanAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+);
+
+// Use the variant value to determine which component to render
+if (variant) {
+    return <FeatureEnabledComponent />
+} else {
+    return <FeatureDisabledComponent />
+}
+```
+
+## JSON Assignments
+
+JSON flags work best for advanced configuration use cases. The JSON flag can include structured information such as 
+
+- the text of a marketing copy for a promotional campaign 
+- the address of a different hero image. 
+
+Using this pattern, a team can make minor changes to the copy and design of a website without having to go through an entire code release process.
+
+For example, the following code shows how to use a JSON flag to configure a campaign.
+
+```js
+import * as EppoSdk from "@eppo/js-client-sdk";
+
+const client = EppoSdk.getInstance();
+
+const flagKey = "flag-key-123";
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = {
+    hero: false,
+    heroImage: "placeholder.png",
+    heroTitle: "Placeholder Hero Title",
+    heroDescription: "Placeholder Hero Description"
+}; 
+
+const campaignJson = client.getJSONAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+);
+
+if (campaignJson !== null) {  
+    campaign.hero = true;
+    campaign.heroImage = campaignJson.heroImage;
+    campaign.heroTitle = campaignJson.heroTitle || "";
+    campaign.heroDescription = campaignJson.heroDescription || "";
+}
+
+// JSX component sets itself based on the campaignJson
+<Hero image={campaignJson.heroImage} title={campaignJson.heroTitle} description={campaignJson.heroDescription} />
+```
+
+## Integer and Numeric Assignments
+
+Integer and numeric assignments work the same way but return either an integer or a floating point number. These assignments are useful where you want to use a numeric value to drive business logic such as pricing on an item or a number of items to display in a list.
+
+```js
+import * as EppoSdk from "@eppo/js-client-sdk";
+
+const client = EppoSdk.getInstance();
+const flagKey = "flag-key-123";
+const subjectKey = getUserId() || "user-123";
+const defaultAssignment = 0; 
+
+// example of getting an integer assignment
+const numberOfItems = client.getIntegerAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+);
+
+// example of getting a numeric assignment
+const price = client.getNumericAssignment(
+  flagKey,
+  subjectKey,
+  defaultAssignment
+);
+
+// use the assignment to drive business logic
+<ItemList items={numberOfItems} />
+
+// use the assignment to drive business logic
+<Item price={price}/>
+
+```
+
+### Assignment Logger Schema
+
+The SDK will invoke the `logAssignment` function with an `assignment` object that contains the following fields:
+
+<ApiOptionRef 
```

Comment:
something here isn't rendering right:

![image](https://github.com/user-attachments/assets/beacd140-15e9-47e3-b4eb-0c8348cf50db)

Reply:
Forgot to import the component. Fixed.